services/thangs_login_service.py

```diff
@@ -3,48 +3,32 @@
 import json
 import webbrowser
 import requests
-import threading
 
-from config import get_config
+from config import set_token, get_bearer_json_file_location, get_api_token
 from api_clients import thangs_login_client
 from time import sleep
 from typing import Optional
-
+from .threading_service import get_threading_service
 
 class ThangsLoginService:
     __GRANT_CHECK_INTERVAL_SECONDS = 0.5  # 500 milliseconds
     __MAX_ATTEMPTS = 600  # 5 minutes worth
 
     def __init__(self):
-        self.__cached_token = None
         self.__login_client = thangs_login_client.ThangsLoginClient()
 
-    def __get_bearer_json_file_location(self) -> str:
-        bearer_location = os.path.join(
-            os.path.dirname(get_config().main_addon_file_location), 'bearer.json')
-        return bearer_location
-
-    def get_api_token(self) -> Optional[str]:
-        if self.__cached_token:
-            return self.__cached_token
-
-        bearer_location = self.__get_bearer_json_file_location()
-        if not os.path.exists(bearer_location):
-            return None
-        bearer_file = open(bearer_location)
-        data = json.load(bearer_file)
-        return data["bearer"]
-
-    def login_user(self, cancellation_event: threading.Event = None) -> None:
-        if self.__cached_token:
-            self.__cached_token = None
+    def login_user(self) -> None:
+        if get_api_token:
+            set_token(None)
 
         challenge_id = uuid.uuid4()
         webbrowser.open(self.__login_client.get_browser_authenticate_url(challenge_id))
 
         attempts = 0
 
-        while attempts < ThangsLoginService.__MAX_ATTEMPTS and not (cancellation_event and cancellation_event.is_set()):
```

Comment:
I couldn't seem to get this part of the line to work as is so I replaced it with the global thread: `and not (cancellation_event and cancellation_event.is_set()):`

Reply:
So why did we switch away from Events and go to a boolean on a global class?  Could we not have had a global cancellation event instead?

I worry because apparently Python is weird and setting a boolean isn't an atomic operation.  https://stackoverflow.com/a/43879450/550062

Reply:
I changed this bool to an event.